### DISCUSSION TITLE
Get seconds since midnight

### DISCUSSION THREAD

**Discussion by iSerganov** [savonet/liquidsoap#4063] [Q&A] [answered]
Hey Community!
Can someone please help me to get seconds since midnight from withing Liquidsoap script?

I am trying to do something like this:
```
let current_time = time()
seconds_since_midnight =  current_time.hour * 3600.0 + current_time.min * 60.0 + current_time.sec
```

Unfortunately not much luck...

Any help is much appreciated.

**Comment by vitoyucepi** [comment, marked as answer]
Hi @iSerganov,
There are two ways to get the seconds since midnight
1. Get the remainder after dividing by a number of seconds in a day after adjusting for the time zone.
   ```ruby
   def seconds_since_midnight()
     current_time = int(time())
     let _.{utc_diff} = time.zone()
     SECONDS_IN_DAY = 86400
     (current_time + utc_diff) mod SECONDS_IN_DAY
   end
   ```
1. Using the local time in the form of a data structure.
   ```ruby
   def seconds_since_midnight()
     let {hour, min, sec} = time.local()
     SECONDS_IN_HOUR = 3600
     SECONDS_IN_MINUTE = 60
     hour * SECONDS_IN_HOUR + min * SECONDS_IN_MINUTE + sec
   end
   ```

**Reply by iSerganov** [reply]
Thank you, actually I've discovered the second option few minutes ago. 
Though the first one seems more interesting to me.
